backend/src/scripts/migrations/4.30/addVertexCountsToMeshBSONs.js

```diff
@@ -34,22 +34,26 @@ const processModel = async (teamspace, model) => {
 	const superMeshes = await find(teamspace, stashDB, { type: 'mesh' }, { m_map: 1 });
 
 	const spanningMeshVerticesCounts = [];
-	const meshUpdates = [];
 
 	// eslint-disable-next-line camelcase
-	superMeshes.forEach(({ m_map }) => {
+	superMeshes.forEach(async ({ m_map }) => {
 		if (m_map.length > 1) {
 			// eslint-disable-next-line camelcase
-			meshUpdates.push(...m_map.map(({ map_id, v_from, v_to }) => {
+			for (let j = 0; j < m_map.length; j++) {
+				// eslint-disable-next-line camelcase
+				const { map_id, v_from, v_to } = m_map[j];
 				// eslint-disable-next-line camelcase
 				const verticesCount = v_to - v_from;
-				return updateOne(
+				// Executing the following `updateOne` concurrently causes
+				// JS heap to run out of memory in some instances
+				// eslint-disable-next-line no-await-in-loop
+				await updateOne(
```

Comment:
I think you just got lucky here.

You're getting all the super meshes, then for each of them, you're, in parallel, doing each updates serially.

So this works if you don't have small amount of super meshes but they have dense m_maps.

but you will run into the same problem if you have a lot of super meshes.

Reply:
I think the problem is too many active promises, instead of having many updateOnes, maybe look into bulkWrite? https://mongodb.github.io/node-mongodb-native/4.9/classes/Collection.html#bulkWrite

Reply:
assuming the version we're using here supports bulkWrite, if not then you'd have to write something to limit the amount of supermeshes you're processing in parallel as well